examples/postgres/custom_arrays/README.md

```diff
@@ -0,0 +1,1162 @@
+            Ok(_) => Ok(true),
+            Err(_) => Ok(false),
+        }
+    }
+```
+
+### Check Service Online
+
+To check if a service is online, we again lean on the generated DSL:
+
+```rust
+    pub fn check_if_service_id_exists(
+        db: &mut Connection,
+        param_service_id: i32,
+    ) -> QueryResult<bool> {
+        match service.find(param_service_id).first::<Service>(db) {
+            Ok(_) => Ok(true),
+            Err(_) => Ok(false),
+        }
+    }
+```
+
+### Get Service Dependencies
+
+Notice, if we select dependencies instead of online, we return all the dependencies of a service. Let’s add another
+method to do just that:
+
+```rust
+    pub fn get_all_service_dependencies(
+        db: &mut Connection,
+        param_service_id: i32,
+    ) -> QueryResult<Vec<Option<i32>>> {
+        service
+            .filter(service_id.eq(param_service_id))
+            .select(dependencies)
+            .first::<Vec<Option<i32>>>(db)
+    }
+```
+
+### Get Service Endpoints
+
+And likewise, if we select endpoints, we get all API endpoints of a service:
+
+```rust
+    pub fn get_all_service_endpoints(
+        db: &mut Connection,
+        param_service_id: i32,
+    ) -> QueryResult<Vec<Option<Endpoint>>> {
+        service
+            .filter(service_id.eq(param_service_id))
+            .select(endpoints)
+            .first::<Vec<Option<Endpoint>>>(db)
+    }
+```
+
+As an observation, when we replace the filter criterion with a test if a service is online, we get all services with
+online set to true and that is very easy to implement:
+
+### Get All Online Services
+
+```rust
+    pub fn get_all_online_services(db: &mut Connection) -> QueryResult<Vec<Self>> {
+        service
+            .filter(online.eq(true))
+            .select(Service::as_returning())
+            .load::<Service>(db)
+    }
+```
+
+### Get All Offline Services
+
+Likewise, we use the same filter to
+find and return all services with online set to false.
+
+```rust
+    pub fn get_all_offline_services(db: &mut Connection) -> QueryResult<Vec<Self>> {
+        service
+            .filter(online.eq(false))
+            .select(Service::as_returning())
+            .load::<Service>(db)
+    }
+```
+
+### Check All Services Online
+
+Finally, we can implement the last method that takes a collection of service ID’s, checks all of them, and returns
+either true, or the service ID that is offline.
+
+```rust
+    pub fn check_all_services_online(
+        db: &mut Connection,
+        services: &[i32],
+    ) -> QueryResult<(bool, Option<String>)> {
+        for id in services {
+            match Service::check_if_service_id_online(db, *id) {
+                Ok(res) => {
+                    if !res {
+                        return Ok((false, Some(format!("Service {} is offline", id))));
+                    }
+                }
+                Err(e) => return Err(e),
+            };
```

Comment:
```suggestion
            let res = Service::check_if_service_id_online(db, *id)?;
            if !res {
                return Ok((false, Some(format!("Service {} is offline", id))));
            }
```

Reply:
That is from check_all_services_online; I've replaced the sample in the Readme with the actual implementation.
Good catch.